storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -538,13 +539,23 @@ static inline void xtrabackup_add_datasink(ds_ctxt_t *ds) {
 }
 
 /* ======== Datafiles iterator ======== */
-datafiles_iter_t *datafiles_iter_new() {
+datafiles_iter_t *datafiles_iter_new(
+    const std::shared_ptr<const xb_dd_tablespaces> dd) {
   datafiles_iter_t *it = new datafiles_iter_t();
 
   mutex_create(LATCH_ID_XTRA_DATAFILES_ITER_MUTEX, &it->mutex);
 
   Fil_iterator::for_each_file([&](fil_node_t *file) {
-    it->nodes.push_back(file);
+    auto space_id = file->space->id;
+    if (dd != nullptr && fsp_is_ibd_tablespace(space_id) &&
+        !fsp_is_system_or_temp_tablespace(space_id) &&
+        !fsp_is_undo_tablespace(space_id) && dd->count(space_id) == 0) {
+      xb::warn() << "Skipping, Tablepspace " << file->name
+                 << " space_id: " << space_id << " does not exists"
+                 << " in INFORMATION_SCHEMA.INNODB_TABLESPACES";
```

Comment:
We should do this orphan check only if lock-ddl=ON? Because if we copied the IBD, with lock-ddl=OFF, I_S query will not have space_id information.

Reply:
and add if (srv_backup_mode = true)?

Reply:
we are building  xb_dd_space only if 
`  if (opt_lock_ddl) {
      xb_dd_space = xb_build_space_id_set(mysql_connection);
      ut_ad(xb_dd_space->size());
    }

Reply:
ok, there could be new users who this function from different places.  An assertion ut_ad(srv_backup_mode) and opt_lock_ddl=ON would be nice